tests/core/query/test_relational_query.py

```diff
@@ -23,6 +23,110 @@ def test_match_table_to_element(sdata_query_aggregation):
     # TODO: add tests for labels
 
 
+def test_left_inner_right_exclusive_join(sdata_query_aggregation):
+    element_dict, table = join_sdata_spatialelement_table(
+        sdata_query_aggregation, "values_polygons", "table", "right_exclusive"
+    )
+    assert table is None
+    assert all(element_dict[key] is None for key in element_dict)
+
+    sdata_query_aggregation["values_polygons"] = sdata_query_aggregation["values_polygons"].drop([10, 11])
+    with pytest.raises(AssertionError, match="No table with"):
+        join_sdata_spatialelement_table(sdata_query_aggregation, "values_polygons", "not_existing_table", "left")
+
+    # Should we reindex before returning the table?
+    element_dict, table = join_sdata_spatialelement_table(sdata_query_aggregation, "values_polygons", "table", "left")
+    assert all(element_dict["values_polygons"].index == table.obs["instance_id"].values)
+
+    # Check no matches in table for element not annotated by table
+    element_dict, table = join_sdata_spatialelement_table(sdata_query_aggregation, "by_polygons", "table", "left")
+    assert table is None
+    assert element_dict["by_polygons"] is sdata_query_aggregation["by_polygons"]
+
+    # Check multiple elements, one of which not annotated by table
+    with pytest.warns(UserWarning, match="The element"):
+        element_dict, table = join_sdata_spatialelement_table(
+            sdata_query_aggregation, ["by_polygons", "values_polygons"], "table", "left"
+        )
+    assert "by_polygons" in element_dict
+
+    # check multiple elements joined to table.
+    sdata_query_aggregation["values_circles"] = sdata_query_aggregation["values_circles"].drop([7, 8])
+    element_dict, table = join_sdata_spatialelement_table(
+        sdata_query_aggregation, ["values_circles", "values_polygons"], "table", "left"
+    )
+    indices = pd.concat(
```

Comment:
Good here. But as implied above it is super crucial that the user knows that this is true only if the initial order between regions and table rows matched, or if the join API is called with `match_rows_order=True`.

Reply:
will clarify in docstring

Reply:
clarified but not with `match_rows` as bool, rather literal